airflow/providers/google/cloud/hooks/cloud_batch.py

```diff
@@ -0,0 +1,229 @@
+
+    ) -> operation.Operation:
+        return self.get_conn().delete_job(name=f"projects/{project_id}/locations/{region}/jobs/{job_name}")
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def list_jobs(
+            self,
+            region: str,
+            project_id: str = PROVIDE_PROJECT_ID,
+            filter: Optional[str] = None,
+            limit: Optional[int] = None) -> Iterable[Job]:
+
+        if limit is not None and limit < 0:
+            raise AirflowException(
+                "The limit for the list jobs request should be greater or equal to zero")
+
+        list_jobs_request: ListJobsRequest = ListJobsRequest(
+            parent=f"projects/{project_id}/locations/{region}",
+            filter=filter
+        )
+
+        jobs: pagers.ListJobsPager = self.get_conn().list_jobs(
+            request=list_jobs_request)
+        
+        return self._limit_list(jobs, limit)
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def list_tasks(
+            self,
+            region: str,
+            job_name: str,
+            project_id: str = PROVIDE_PROJECT_ID,
+            group_name: str = 'group0',
+            filter: Optional[str] = None,
+            limit: Optional[int] = None) -> Iterable[Task]:
+
+        if limit is not None and limit < 0:
+            raise AirflowException(
+                "The limit for the list tasks request should be greater or equal to zero")
+
+        list_tasks_request: ListTasksRequest = ListTasksRequest(
+            parent=f"projects/{project_id}/locations/{region}/jobs/{job_name}/taskGroups/{group_name}",
+            filter=filter
+        )
+
+        tasks: pagers.ListTasksPager = self.get_conn().list_tasks(
+            request=list_tasks_request)
+
+        return self._limit_list(tasks, limit)
+
+    def wait_for_job(
+            self,
+            job_name: str,
+            polling_period_seconds: float = 10,
+            timeout: Union[float, None] = None
+    ) -> Job:
+        client = self.get_conn()
+        while timeout is None or timeout > 0:
+            try:
+                job = client.get_job(name=f"{job_name}")
+                status: JobStatus.State = job.status.state
+                if status == JobStatus.State.SUCCEEDED \
+                        or status == JobStatus.State.FAILED \
+                        or status == JobStatus.State.DELETION_IN_PROGRESS:
+                    return job
+                else:
+                    sleep(polling_period_seconds)
+            except Exception as e:
+                self.log.exception(
+                    "Exception occurred while checking for job completion.")
+                raise e
+
+            if timeout is not None:
+                timeout -= polling_period_seconds
+
+        raise AirflowException(f"Job with name [{job_name}] timed out")
+
+    def get_job(self, job_name) -> Job:
+        return self.get_conn().get_job(name=f"{job_name}")
```

Comment:
```suggestion
        return self.get_conn().get_job(name=job_name)
```

Also present in other places